x-pack/plugins/security_solution/server/endpoint/services/actions/mocks.ts

```diff
@@ -20,40 +20,66 @@ import {
   ENDPOINT_ACTIONS_INDEX,
 } from '../../../../common/endpoint/constants';
 
-export const createActionRequestsEsSearchResultsMock =
-  (): estypes.SearchResponse<LogsEndpointAction> => {
-    const endpointActionGenerator = new EndpointActionGenerator('seed');
+export const createActionRequestsEsSearchResultsMock = (
+  agentIds?: string[]
+): estypes.SearchResponse<LogsEndpointAction> => {
+  const endpointActionGenerator = new EndpointActionGenerator('seed');
 
-    return endpointActionGenerator.toEsSearchResponse<LogsEndpointAction>([
-      endpointActionGenerator.generateActionEsHit({
-        EndpointActions: { action_id: '123' },
-        agent: { id: 'agent-a' },
-        '@timestamp': '2022-04-27T16:08:47.449Z',
-      }),
-    ]);
-  };
+  return endpointActionGenerator.toEsSearchResponse<LogsEndpointAction>([
+    endpointActionGenerator.generateActionEsHit({
+      EndpointActions: { action_id: '123' },
+      agent: { id: agentIds ? agentIds : 'agent-a' },
+      '@timestamp': '2022-04-27T16:08:47.449Z',
+    }),
+  ]);
+};
 
-export const createActionResponsesEsSearchResultsMock = (): estypes.SearchResponse<
-  LogsEndpointActionResponse | EndpointActionResponse
-> => {
+export const createActionResponsesEsSearchResultsMock = (
+  agentIds?: string[]
+): estypes.SearchResponse<LogsEndpointActionResponse | EndpointActionResponse> => {
   const endpointActionGenerator = new EndpointActionGenerator('seed');
   const fleetActionGenerator = new FleetActionGenerator('seed');
 
-  return endpointActionGenerator.toEsSearchResponse<
-    LogsEndpointActionResponse | EndpointActionResponse
-  >([
+  let hitSource: Array<
+    | estypes.SearchHit<EndpointActionResponse>
+    | estypes.SearchHit<LogsEndpointActionResponse<object>>
+  > = [
     fleetActionGenerator.generateResponseEsHit({
       action_id: '123',
       agent_id: 'agent-a',
       error: '',
       '@timestamp': '2022-04-30T16:08:47.449Z',
     }),
     endpointActionGenerator.generateResponseEsHit({
-      agent: { id: 'agent-a' },
+      agent: { id: agentIds ? agentIds : 'agent-a' },
```

Comment:
I did not realize that the Endpoint Response record had the `agent.id` set as `string | string[]`. 

Is this really true? can we validate the index is set that way?

Does not make sense to me as to why it's an array of string. The response will always just be for a single endpoint agent

cc/ @pzl

Reply:
you know, funny thing about elasticsearch: it implicitly allows any field to also be an array of that type.

If you wanted to store an array of favorite color names, the mapping would *not* be `favourite_colors: keyword[]`.  But rather `favourite_colors: keyword`. ES generally allows any field to also be an array.


If one were to send a mass-action (with multiple agent_ids as the targets), I'm not sure what would come back. Endpoint generally parrots the input action back as-it-was in the response doc. I don't actually recall if it would receive all the target agent_ids, or just the one that it is.

The amount of things that are going to break when someone finally tests a mass-action.. phew..

Reply:
Thanks @pzl 

@ashokaditya - I would propose that you don't make this change in the generator and suggest that we change the Type for the response so that it is only `string` not `string[]`

Reply:
What about this one? Can we leave response mock here showing only one agent id?

Reply:
I guess we can. I made this change to use the mocked data in the tests. I'll try and do the test without this change.